packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -142,20 +146,51 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         return "";
     }
 
-    public async createBlob(file: Uint8Array): Promise<api.ICreateBlobResponse> {
-        this.checkSnapshotUrl();
+    public async createBlob(file: IsoBuffer): Promise<api.ICreateBlobResponse> {
+        this.checkAttachmentPOSTUrl();
 
-        return PerformanceEvent.timedExecAsync(
-            this.logger,
-            {
-                eventName: "createBlob",
-                size: file.length,
-            },
-            async () => {
-                // Future implementation goes here
-                // Need to wrap implementation with getWithRetryForTokenRefresh()
-                throw new Error("StandardDocumentStorageManager.createBlob() not implemented");
-            });
+        const response = await getWithRetryForTokenRefresh(async (options) => {
+            const storageToken = await this.getStorageToken(options, "CreateBlob");
+            const { url, headers } = getUrlAndHeadersWithAuth(`${this.attachmentPOSTUrl}/content`, storageToken);
+            headers["Content-Type"] = "application/octet-stream";
+
+            return PerformanceEvent.timedExecAsync(
+                this.logger,
+                {
+                    eventName: "createBlob",
+                    size: file.length,
+                },
+                async () => fetchHelper<api.ICreateBlobResponse>(
+                    url,
+                    {
+                        body: file,
+                        headers,
+                        method: "POST",
+                    },
+                ),
+            );
+        });
+
+        return response.content;
+    }
+
+    public async readBlob(blobid: string) {
+        this.checkAttachmentGETUrl();
+
+        const response = await getWithRetryForTokenRefresh(async (options) => {
+            const storageToken = await this.getStorageToken(options, "ReadDataBlob");
+            const { url, headers } = getUrlAndHeadersWithAuth(`${this.attachmentGETUrl}/${blobid}/content`, storageToken);
```

Comment:
Should we wrap blobId with encodeURIComponent() ? I.e. can it has special symbols that are not allowed in URI?

Reply:
I don't think it's necessary for SPO blob IDs and we don't encode them elsewhere in the SPO driver

Reply:
It is worth confirming that with SPO. Note that today (strictly from usage POV) we do not use any workflows where IDs are in URI - we download while snapshot (not individual parts) and past whole snapshot, i.e. IDs are always part of payload, not on URI. Even if we have some code that has IDs on URIs, it's not used today, so no guarantee that code is correct / works.
I'd say for safety, it's better to always wrap it with encodeURIComponent()